packages/shared/src/utils/storage/storage_events.js

```diff
@@ -1,11 +1,17 @@
-export const setStorageEvents = () => {
+export const setStorageEvents = root_store => {
     window.addEventListener('storage', evt => {
         switch (evt.key) {
             case 'active_loginid':
                 if (document.hidden) {
                     window.location.reload();
                 }
                 break;
+            case 'reality_check_dismissed':
+                if (document.hidden) {
+                    // if new value is true, hide reality check, otherwise show it
+                    root_store.client.setVisibilityRealityCheck(!JSON.parse(evt.newValue));
+                }
+                break;
```

Comment:
Instead of getting the root store, can we instead use events (like we do with service worker updates)?
P.S.: We can actually move the `setStorageEvents` function into Core. This function isn't (and doesn't need to be) called by any other part of the app. It's only used once and set by Core.